lib/plugins/wallet/galoy/galoy.js

```diff
@@ -0,0 +1,364 @@
+          path
+        }
+        invoice {
+          paymentRequest
+        }
+      }
+    }`,
+    'variables': { 'input': { 'walletId': `${walletId}`, 'amount': `${cryptoAtoms}` } }
+  }
+  return request(createInvoice, token)
+    .then(result => {
+      return result.data.lnInvoiceCreate.invoice.paymentRequest
+    })
+    .catch(err => {
+      throw err
+    })
+}
+
+function balance (account, cryptoCode, settings, operatorId) {
+  const externalCryptoCode = coinUtils.getExternalCryptoCode(cryptoCode)
+  return checkCryptoCode(cryptoCode)
+    .then(() => getGaloyAccount(account.apiKey))
+    .then(galoyAccount => {
+      // account has a list of wallets, should we consider the balance of each one?
+      // for now we'll get the first BTC wallet that matches the defaultWalletId
+      const wallet = _.head(
+        _.filter(wallet => wallet.walletCurrency === externalCryptoCode &&
+          wallet.id === galoyAccount.defaultWalletId &&
+          wallet.id === account.walletId)(galoyAccount.wallets)
+      )
+      return new BN(wallet.balance || 0)
+    })
+    .catch(err => {
+      throw err
+    })
+}
+
+function newAddress (account, info, tx, settings, operatorId) {
+  const { cryptoAtoms, cryptoCode } = tx
+  const externalCryptoCode = coinUtils.getExternalCryptoCode(cryptoCode)
+  return checkCryptoCode(cryptoCode)
+    .then(() => getGaloyAccount(account.apiKey))
+    .then(galoyAccount => {
+      const wallet = _.head(
+        _.filter(wallet => wallet.walletCurrency === externalCryptoCode &&
+          wallet.id === galoyAccount.defaultWalletId &&
+          wallet.id === account.walletId)(galoyAccount.wallets)
+      )
+      const promises = [
+        newOnChainAddress(wallet.id, account.apiKey),
+        newInvoice(wallet.id, cryptoAtoms, account.apiKey)
+      ]
+      return Promise.all(promises)
+    })
+    .then(([onChainAddress, invoice]) => {
+      return `bitcoin:${onChainAddress}?amount=${cryptoAtoms}&lightning=${invoice}`
+    })
+}
+
+function getStatus (account, tx, requested, settings, operatorId) {
+  const { toAddress, cryptoAtoms, cryptoCode } = tx
+  const mapStatus = tx => {
+    if (!tx) return 'notSeen'
+    if (tx.node.status === TX_PENDING) return 'authorized'
+    if (tx.node.status === TX_SUCCESS) return 'confirmed'
+    return 'notSeen'
+  }
+  const externalCryptoCode = coinUtils.getExternalCryptoCode(cryptoCode)
+  const address = coinUtils.parseUrl(address)
+  return checkCryptoCode(cryptoCode)
+    .then(() => getGaloyAccount(account.apiKey))
+    .then(galoyAccount => {
+      const wallet = _.head(
+        _.filter(wallet => wallet.walletCurrency === externalCryptoCode &&
+          wallet.id === galoyAccount.defaultWalletId &&
+          wallet.id === account.walletId)(galoyAccount.wallets)
+      )
+      const transactions = wallet.transactions.edges
+      if (isLightning(address)) {
+        const paymentHash = invoice.decode(address).paymentHash.toString('hex')
```

Comment:
Doesn't this `invoice.decode()` ever fail?

Reply:
yes, in that case, the function will throw an error